k1run.sh

```diff
@@ -0,0 +1,176 @@
+    echo "Usage: sh k1run.sh [folder] [option] [option2]"
+    echo "Script that automate managing docker."
+    echo
+    echo "If no parameter is passed then boot and initialize the site."
+    echo
+    echo "--build    Start the site and initialize the site."
+    echo "--down     Stop the site. Keep data"
+    echo "--destroy  Stop the site, destory data"
+    echo "--reboot   Restart the site - destroy all containers and re-initialize"
+    echo "--load     Reload the site, use existing data"
+    echo "--phpunit  Initialize for phpunit tests"
+    echo "--behat    Initialize for behat tests"
+}
+
+exists_in_list() {
+    LIST=$1
+    DELIMITER=$2
+    VALUE=$3
+    LIST_WHITESPACES=`echo $LIST | tr "$DELIMITER" " "`
+    for x in $LIST_WHITESPACES; do
+        if [ "$x" = "$VALUE" ]; then
+            return 1
+        fi
+    done
+    return 0
+}
+
+if [ $# -eq 0 ];  then
+    echo "No arguments supplied"
+    help_messages
+    return
+fi
+
+#Count the variables passed in.
+variablecount=$#
+if [ $# -lt 2 ] || [ $# -gt 3 ] ;  then
+    echo "Invalid number of arguments passed in. Must be between 2 and 3 arguments"
+    help_messages
+    return
+fi
+
+cwd=$( dirname "$PWD" )
+
+folder="${1}"
+
+folder="${cwd}/${folder}"
+if [ ! -d "${folder}" ]; then
+   echo "${folder} is not valid"
+   return
+fi
+
+SWITCH=$2
+SWITCH2=$3
+
+# Variables
+# MOODLE_DOCKER_DB - database used by Moodle - default maria db.
+# MOODLE_DOCKER_WWWROOT - folder where the Moodle code is located;
+# MOODLE_DOCKER_PHP_VERSION - version of PHP used by Moodle. Default 8.1.
+
+if [ "$SWITCH" = "--help" ]; then
+    help_messages
+fi
+
+# Check to see if the options are valied.
+
+list_of_options="--build --down --destroy --reboot --load --phpunit --behat"
+
+if  exists_in_list "$list_of_options" " " $SWITCH;  then
+    echo "Invalid option $SWITCH"
+    help_messages
+    return
+fi
+
+if [ "$variablecount" -eq 3 ]; then
+    if  exists_in_list "$list_of_options" " " $SWITCH2;  then
+        echo "Invalid option $SWITCH2"
+        help_messages
+        return
+    fi
+fi
+
+# Always use mariadb as a database.
+
+export MOODLE_DOCKER_DB=mariadb
+export MOODLE_DOCKER_WWWROOT=${folder}
+
+# Check the Moodle version. If its 4.5 then set php version to 8.3
+moodlever=$(grep "$branch   = '405';" $folder/version.php)
+if [ "$moodlever" ]; then
+   export MOODLE_DOCKER_PHP_VERSION=8.3
+fi
+
+# Use the local.yml_single for one site - includes adminer.
+cp local.yml_single local.yml
+cp config.docker-template.php $MOODLE_DOCKER_WWWROOT/config.php
+
+# Build
+if [ "$SWITCH" = "--build" ]; then
+    # Start up containers
+    bin/moodle-docker-compose up -d
+    # Wait for DB to come up
+    bin/moodle-docker-wait-for-db
+    # Initialize the database
+    bin/moodle-docker-compose exec webserver php admin/cli/install_database.php --agree-license --fullname="K1MOODLE" --shortname="K1MOODLE" --summary="K1 Moodle dev" --adminpass="test" --adminemail="admin@example.com"
```

Comment:
Shouldn't we call destroy first to force delete the container? up will only build if the container do not exist yet. If they exist, it will up them.

https://github.com/moodlehq/moodle-docker?tab=readme-ov-file#quick-start
\# Shut down and destroy containers
bin/moodle-docker-compose down

Reply:
I put a check in to see if the containers are running. If they are then this stops. I don't want someone to accidently delete containers by mistake